hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/RecordLevelIndexSupport.scala

```diff
@@ -91,6 +93,23 @@ class RecordLevelIndexSupport(spark: SparkSession,
   def isIndexAvailable: Boolean = {
     metadataConfig.isEnabled && metaClient.getTableConfig.getMetadataPartitions.contains(HoodieTableMetadataUtil.PARTITION_NAME_RECORD_INDEX)
   }
+
+  /**
+   * Returns true if the query type is supported by the index.
+   */
+  override def supportsQueryType(options: Map[String, String]): Boolean = {
+    options.get(TIME_TRAVEL_AS_OF_INSTANT.key)
```

Comment:
Should we have the first check on query type? A very rare case might occur where user uses `as.of.instant` with RO query and still allow query to go through.

Reply:
makes sense, updated the condition